src/robotFactory.js

```diff
@@ -1,10 +1,79 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(y = 1) {
+    this.coords.y += y;
+  }
+
+  goBack(y = 1) {
+    this.coords.y -= y;
+  }
+
+  goRight(x = 1) {
+    this.coords.x += x;
+  }
+
+  goLeft(x = 1) {
+    this.coords.x -= x;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, `
+    + `Chip version: ${this.chipVersion}, `
+    + `Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords.z = coords.z || 0;
+  }
+  goUp(z = 1) {
+    this.coords.z += z;
+  }
+
+  goDown(z = 1) {
+    this.coords.z -= z;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion,
+    maxLoadWeight,
+    currentLoad = null) {
+    super(name, weight, coords, chipVersion);
+
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad;
+  }
+  hookLoad(cargo) {
+    if (!this.currentLoad
+    && cargo.weight <= this.maxLoadWeight) {
```

Comment:
we use this approach if we want to move the if condition to the new line
```suggestion
    if (
      !this.currentLoad
      && cargo.weight <= this.maxLoadWeight
    ) {
```